### PR TITLE
GODRIVER-1589 consolidate pool.drain and pool.clear

### PR DESCRIPTION
**Pull request by iwysiu**
There were some points where we were calling drain when we shouldn't.

### CODE REVIEW

File: x/mongo/driver/topology/resource_pool_test.go [hunk 1]
```diff
@@ -182,6 +182,7 @@ func TestResourcePool(t *testing.T) {
 			assert.Equal(t, int32(5), expiredCalled, "expected expire to be called 5 times, got %v", expiredCalled)
 			closeCalled := ec.getCloseCalled()
 			assert.Equal(t, int32(3), closeCalled, "expected close to be called 3 times, got %v", closeCalled)
+			assert.True(t, rp.maintainTimer.Stop(), "expected timer to be reset")
```
Comment (divjotarora) [inline x/mongo/driver/topology/resource_pool_test.go:185]: Just to make sure I'm understanding this correctly: the `Maintain` function happens after the timer fires. It does the maintenance and then restarts `maintainTimer`. The `time.Timer.Stop` function will return false if the timer has already fired, so asserting that `Stop` returns true here means that we've ensured `Maintain` correctly reset it.

If this is correct, can we test this in a less roundabout way by actually making `Maintain` happen twice? If not, can you add a comment here explaining why this check is needed?

Comment (iwysiu) [reply x/mongo/driver/topology/resource_pool_test.go:187]: comment added. The issue I was running into was proving that Maintain was running twice, which is why I did this instead

File: x/mongo/driver/topology/server.go [hunk 2]
```diff
@@ -469,16 +470,6 @@ func (s *Server) updateDescription(desc description.Server, initial bool) {
 		c <- desc
 	}
 	s.subLock.Unlock()
-
-	if initial {
-		// We don't clear the pool on the first update on the description.
-		return
-	}
-
-	switch desc.Kind {
```
Comment (craiggwilson) [inline x/mongo/driver/topology/server.go:478]: Is there no other time, other than when we manually set the kind to unknown on line 258, that the kind becomes unknown and the pool should be cleared?

Comment (iwysiu) [reply x/mongo/driver/topology/server.go:478]: a lot of the places that should clear when the kind is set to unknown were actually double clearing before by calling both updateDescription (which drained) and pool.clear, so now those places are only clearing once

**Pull request merged**